### PR TITLE
GSAGH-450-Section-Pools

### PR DESCRIPTION
**Pull request by tlmnrnhrdt**

### CODE REVIEW

File: GsaGH/Components/4_Analysis/CreateSteelSectionPool.cs [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    protected override void RegisterOutputParams(GH_OutputParamManager pManager) {
+      pManager.AddParameter(new GsaSectionParameter());
+    }
+
+    protected override void SolveInstance(IGH_DataAccess da) {
+      int pid = 0;
+      da.GetData(0, ref pid);
+
+
+      GsaSectionGoo sectionGoo = null;
```
Comment (kpne) [inline GsaGH/Components/4_Analysis/CreateSteelSectionPool.cs:37]:
```suggestion
      da.GetData(0, ref pid);

      GsaSectionGoo sectionGoo = null;
```

File: GsaGH/Components/4_Analysis/CreateSteelSectionPool.cs [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+      if (da.GetData(1, ref sectionGoo)) {
+        if (sectionGoo is GsaSectionGoo) {
+          var section = (GsaSectionGoo)sectionGoo.Duplicate();
+          section.Value.ApiSection.Pool = pid;
+          da.SetData(0, section);
+        }
```
Comment (kpne) [inline GsaGH/Components/4_Analysis/CreateSteelSectionPool.cs:43]:
```suggestion
      da.GetData(1, ref sectionGoo);
      var section = (GsaSectionGoo)sectionGoo.Duplicate();
      section.Value.ApiSection.Pool = pid;
      da.SetData(0, section);
```

File: GsaGH/Components/4_Analysis/SteelSectionPoolNames.cs [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+namespace GsaGH.Components {
+  public class SteelSectionPoolNames : GH_OasysComponent {
+    public override Guid ComponentGuid => new Guid("6df7c85e-f629-47fd-ab38-7431407777e3");
+    public override GH_Exposure Exposure => GH_Exposure.quinary | GH_Exposure.obscure;
+    public override OasysPluginInfo PluginInfo => GsaGH.PluginInfo.Instance;
+    protected override Bitmap Icon => Resources.SteelSectionPoolNames;
+
+    public SteelSectionPoolNames() : base("Steel Section Pool Names", "Pool Names",
+      "Get or set the Steel Section Pool Names of a GSA Model", CategoryName.Name(),
+      SubCategoryName.Cat4()) {
```
Comment (kpne) [inline GsaGH/Components/4_Analysis/SteelSectionPoolNames.cs:24]: Should this not be in the Model subcategory?

File: GsaGH/Components/4_Analysis/SteelSectionPoolNames.cs [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+            string type = wrapper.Value.GetType().ToString();
+            this.AddRuntimeError("Unable to convert input parameter of type " + type + " to Name");
+            return;
+          }
+          i++;
+        }
+      }
+
+      if (ids.Count > 0) {
+        model.Model.SetSteelSectionPools(new ReadOnlyDictionary<int, string>(sectionPools));
```
Comment (kpne) [inline GsaGH/Components/4_Analysis/SteelSectionPoolNames.cs:94]: do we not need to clone the model?

**Review by kpne — CHANGES_REQUESTED** [review]

File: GsaGH/Components/4_Analysis/SteelSectionPoolNames.cs (hunk 4 above)
Comment (tlmnrnhrdt) [reply GsaGH/Components/4_Analysis/SteelSectionPoolNames.cs:94]: it´s being cloned in line 52

File: GsaGH/Components/4_Analysis/SteelSectionPoolNames.cs (hunk 3 above)
Comment (tlmnrnhrdt) [reply GsaGH/Components/4_Analysis/SteelSectionPoolNames.cs:24]: Where do you wanna put it exactly?

Comment (kpne) [reply GsaGH/Components/4_Analysis/SteelSectionPoolNames.cs:24]: Thinking either of these two:
image

**Review by kpne — APPROVED** [review]

**Pull request merged**